src/escape.rs

```diff
@@ -118,3 +163,73 @@ pub fn escape_html(ob: &mut String, s: &str, secure: bool) {
     }
     ob.push_str(&s[mark..]);
 }
+
+fn parse_entity(s: &str) -> Option<(usize, &'static str)> {
```

Comment:
About 90% sure these don't belong here, entity and literal parsing should be done during parsing.